src/components/gamespage/GamesPageProgressSection/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import "./index.css";
+import { useTranslation } from "react-i18next";
+import { Progress, Row, Col } from "antd";
+import { QuestionOutlined } from "@ant-design/icons";
+import recycleIcon from "../../../images/recycle.svg";
+
+const ProgressSection = ({
+  // XP stands for experience
+  // number
+  currentXP = 50,
+  // svg file
+  prevRewardIcon = recycleIcon,
+  // string
+  prevRewardDescription = "Recycle",
+  // number
+  currentLevel = 5,
+}) => {
+  const nextLevel = currentLevel + 1;
+  const [t] = useTranslation();
+  return (
+    <div className="progressSectionContainer">
+      <Row justify="center">
+        <h3 className="progressSectionContainer___xp">
+          {currentXP} / 100 {t("gamesPage.experiencePoint")}
+        </h3>
+      </Row>
+      <Row justify="center" className="progressSectionContainer___row">
+        <Col xs={6} md={{ push: 1 }} lg={{ push: 2 }}>
+          <h3 className="progressSectionContainer___row___header progressSectionContainer___row___curLevel">
+            {t("gamesPage.level")} {currentLevel}
+          </h3>
+        </Col>
+        <Col xs={12}>
+          <Progress
+            percent={currentXP}
+            className="progressSectionContainer___row___progressBar"
+            strokeColor="#4BA747"
+            trailColor="white"
+            strokeWidth={20}
+            showInfo={false}
+          />
+        </Col>
+        <Col xs={6} md={{ pull: 1 }} lg={{ pull: 2 }}>
+          <h3 className="progressSectionContainer___row___header progressSectionContainer___row___nextLevel">
+            {t("gamesPage.level")} {nextLevel}
+          </h3>
+        </Col>
+      </Row>
+      <Row className="progressSectionContainer___row">
+        <Col span={3} push={5}>
+          <h3 className="progressSectionContainer___row___header">
+            {t("gamesPage.previous")}
+            <br />
+            {t("gamesPage.reward")}
+          </h3>
+          <div className="progressSectionContainer___row___prevRewardDiv">
+            <img
+              src={prevRewardIcon}
+              alt={prevRewardDescription}
+              className="progressSectionContainer___row___prevRewardDiv___reward"
+            />
+          </div>
+          <h3 className="progressSectionContainer___row___header">
+            {prevRewardDescription} {t("gamesPage.badge")}
+          </h3>
+        </Col>
+        <Col span={3} push={13}>
+          <h3 className="progressSectionContainer___row___header">
+            {t("gamesPage.next")}
+            <br />
+            {t("gamesPage.reward")}
+          </h3>
+          <div className="progressSectionContainer___row___header___questionMark">
+            <QuestionOutlined />
+          </div>
+
+          <h3 className="progressSectionContainer___row___header">
+            {t("gamesPage.unknown")}
```

Comment:
It is translation of word "unknown" but instead of renaming it I added a comment to give example. Because I couldn't think of a precise variable name. Names like `unknownWord `didn't feel right. But I'm open for suggestions.

Reply:
> it's not about whether it works for 3 languages or not, it's whether its the correct way to use translations or not. <level word> <number> is a hack, and also a bad translation practice. for example if i worked at an company and i asked you to do this in a task with translations as a take-home task (this is actually something that my students have been asked to do so i'm not making this up either), and i saw this, i would be very turned off. in terms of this particular feature i would prefer a candidate who understands that languages are not guaranteed in order and/or how to leverage translation libraries to adapt to that it's not hard to change this either; it'll be the same amount of code +2 lines https://www.i18next.com/translation-function/interpolation
> […](#)
> On Wed, Nov 18, 2020 at 4:10 AM Tarkan Mansuroğlu ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In src/components/gamespage/GamesPageProgressSection/index.jsx <[#65 (comment)](https://github.com/ReCoded-Org/capstone-istanbul-juniper/pull/65#discussion_r525629253)> : > + // number + currentLevel = 5, +}) => { + const nextLevel = currentLevel + 1; + const [t] = useTranslation(); + return ( + <div className="progressSectionContainer"> + <Row justify="center"> + <h3 className="progressSectionContainer___xp"> + {currentXP} / 100 {t("gamesPage.experiencePoint")} + </h3> + </Row> + <Row justify="center" className="progressSectionContainer___row"> + <Col xs={6} md={{ push: 1 }} lg={{ push: 2 }}> + <h3 className="progressSectionContainer___row___header progressSectionContainer___row___curLevel"> + {t("gamesPage.level")} {currentLevel} I actually created something like <level word> <number>. seviye 1. is not natural but seviye 1 is. I also talked with Ibrahim and said order is same for Arabic. If it is ok I will improve naming keep the code as it is. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#65 (comment)](https://github.com/ReCoded-Org/capstone-istanbul-juniper/pull/65#discussion_r525629253)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAJ6G2JZL7XXLWZXVGSYJD3SQMNGTANCNFSM4TYDALOQ> .

I didn't know about translation practices. Good point. I wasn't feeling lazy or thinking it is difficult. Since I couldn't see the benefit of your suggestion, I didn't want to implement it.

Reply:
why would someone's progress be unknown is what I'm confused about. if they don't have progress they just have 0 exp no?

Reply:
the comment is not explaining what is unknown